app/scripts/modules/core/src/pipeline/config/graph/PipelineGraph.tsx

```diff
@@ -272,6 +274,12 @@ export class PipelineGraph extends React.Component<IPipelineGraphProps, IPipelin
     }
     newState.maxLabelWidth = Math.max(this.minLabelWidth, newState.maxLabelWidth);
 
+    if (newState.maxLabelWidth === this.minLabelWidth) {
+      newState.graphWidth = (newState.phaseCount + 1) * (newState.maxLabelWidth + phaseOffset) + 5 + 'px';
```

Comment:
I have zero context about any of this graph layout code so I apologize for my confusion — are there specific cases/flows where this condition applies? And what's different behaviorally between setting this pre-calculated fixed width and using `100%`?

![math](https://media.giphy.com/media/26xBI73gWquCBBCDe/giphy.gif)

Reply:
I'm wondering the same thing as Erik.  How does this work?  When is it triggered?
making any changes to this code makes me really nervous 😆 are there dragons here?

Is this to allow the horizontal scrolling?  It's a pretty minimal change, but I wonder if css could be applied to the outer element instead?  Either way, we should merge these changes as-is to unblock any users, and come back to it in another PR if there's a better approach landed upon.

Reply:
@christopherthielen @erikmunson I am actually putting this code back in if that makes you feel any better :) I originally took it out, set the width to 100% and made the containing div responsive. 

The reason I added this back is because of how the svg behaves. With just setting the width to `100%` it truncates the graph. I ended up appling the css to the outer element to enable that to be flex w/ scroll, but without explicitly setting the width the image is usually defaulted to a very small size.

Reply:
> I am actually putting this code back in if that makes you feel any better :)

HAHA 👍  🤣  ok! 🌮

Reply:
✅✅✅ WOW so glad we had this chat

Reply:
LOL Unclear if there are now old dragons back in business 🐲 🤣